control/trajectory_follower/src/mpc.cpp

```diff
@@ -217,6 +217,37 @@ void MPC::setReferenceTrajectory(
     }
   }
 
+  /* extend terminal points.
+   * the original trajectory cannot take into account the pose from the position alone
+   * due to smoothing, etc.
+   */
+  if (enable_traj_extending) {
```

Comment:
Please move thses code into a separate function.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2447/commits/c0cf295e6c81d5fd4382415f3096a18da30f9ac5